pkg/tasks/probes.go

```diff
@@ -636,3 +650,76 @@ func detectCCMMigrationStatus(s *state.State) (*state.CCMStatus, error) {
 
 	return status, nil
 }
+
+// detectOpenStackClusterName is used to detect the value that should be passed
+// to the OpenStack CCM via --cluster-name flag.
+//
+// This function is needed to ensure backwards compatibility because initially
+// we didn't set this flag, in which case it defaults to 'kubernetes'.
+// This can cause issues if there are multiple clusters in the same tenant
+// (e.g. Load Balancers with the same name in different clusters will share
+// the same Octavia LB).
+// Changing the --cluster-name causes the CCM to lose all references to the
+// Load Balancers on OpenStack, because the cluster name is used as part of
+// the reference to the LB.
+//
+// The function works in the following way:
+//   * do nothing if the cluster is not an OpenStack cluster
+//   * if it's a new cluster, return the KubeOne cluster name
+//   * if it's an existing cluster:
+//      * if cluster is running in-tree cloud provider: return the KubeOne cluster name
+//      * if cluster is running external cloud provider: check is there
+//        `--cluster-name flag on the OpenStack CCM. If there is, read the
+//        value and use it, otherwise return empty string (defaults to
+//        `kubernetes`)
+//   * if cluster is migrated to external CCM, use the KubeOne cluster name
```

Comment:
Nitpick: This documentation of `detectOpenStackClusterName` doesn't seem to be in line with the actual's function return type, which is `error`. It suggests - at least to me - that this returns a string.

Reply:
@embik Good point, I actually thought about it as well, but I was too lazy to change it. PTAL now.